src/main/java/ca/uhn/fhir/jpa/starter/dotbase/operations/DocumentOperation.java

```diff
@@ -0,0 +1,90 @@
+package ca.uhn.fhir.jpa.starter.dotbase.operations;
+
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.hl7.fhir.r4.model.IdType;
+import org.hl7.fhir.r4.model.ListResource;
+import org.hl7.fhir.r4.model.Meta;
+import org.hl7.fhir.r4.model.Parameters;
+import org.hl7.fhir.r4.model.StringType;
+import org.hl7.fhir.r4.model.DateTimeType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import ca.uhn.fhir.jpa.api.dao.IFhirResourceDao;
+import ca.uhn.fhir.jpa.searchparam.SearchParameterMap;
+import ca.uhn.fhir.jpa.starter.dotbase.DotbaseProperties;
+import ca.uhn.fhir.jpa.starter.dotbase.utils.BeanUtils;
+import ca.uhn.fhir.jpa.starter.dotbase.utils.fhir.ExtensionUtils;
+import ca.uhn.fhir.rest.api.SortOrderEnum;
+import ca.uhn.fhir.rest.api.SortSpec;
+import ca.uhn.fhir.rest.api.server.IBundleProvider;
+import ca.uhn.fhir.rest.param.ReferenceParam;
+import ca.uhn.fhir.rest.param.TokenParam;
+
+@Service
+public class DocumentOperation {
+
+  @Autowired
+  DotbaseProperties dotbaseProperties;
+
+  public Parameters lastEdited(IdType documentId) {
+    ListResource latestSection = this.getLatestDocumentSection(documentId);
+
+    StringType user = this.getUsername(latestSection);
+    DateTimeType lastUpdated = new DateTimeType(latestSection.getMeta().getLastUpdated());
+
+    Parameters params = new Parameters();
+    params.addParameter("username", user);
+    params.setParameter("lastUpdated", lastUpdated);
+
+    return params;
+  }
+
+  private StringType getUsername(ListResource latestSection) {
+    Meta meta = latestSection.getMeta();
+    String version = meta.getVersionIdElement().toString();
+
+    if (version.equals("1")) {
+      throw new Error("No editor exists, document was not edited yet.");
+    }
+    return this.getResourceEditor(meta);
+  }
+
+  private StringType getResourceEditor(Meta meta) {
+    String editorExtension = dotbaseProperties.getResourceUrls().getStructureDefinitionResourceEditor();
+    return (StringType) ExtensionUtils.getExtensionByUrl(meta, editorExtension).getValue();
+  }
+
+  private ListResource getLatestDocumentSection(IdType documentId) {
+    IBundleProvider sectionSearch = this.searchLatestDocumentSection(documentId);
+    IBaseResource latestSection = sectionSearch.getAllResources().get(0);
+
+    if (sectionSearch.isEmpty()) {
+      String isEmpty = String.format("No sections for document with id %s exist.", documentId.getId());
+      throw new Error(isEmpty);
+    }
```

Comment:
Won't this result in an error if the sectionSearch has no results? Either way, I would have expected the guard clause `if (sectionSearch.isEmpty()` to be executed before retrieving the search results.

```suggestion
    if (sectionSearch.isEmpty()) {
      String isEmpty = String.format("No sections for document with id %s exist.", documentId.getId());
      throw new Error(isEmpty);
    }

    IBaseResource latestSection = sectionSearch.getAllResources().get(0);
```

Reply:
Yes thanks, that correct. I moved the it downwards